app/models/product_import/entry_validator.rb

```diff
@@ -62,6 +62,11 @@ def supplier_validation(entry)
         return
       end
 
+      unless Enterprise.find_by_name(supplier_name).is_primary_producer?
```

Comment:
Do you really need to load the enterprise from the database or is it in `@editable_enterprises` already? Ah, it's detour, you figured that yourself. :-)